### PR TITLE
[tests-only][full-ci] added e2e test for adding banned password for public link

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds the test for adding the banned password for public link. When adding the banned passwords then the user should see an error message `Unfortunately, your password is commonly used. please pick a harder-to-guess password for your safety`
Used the existing password in banned-paswword-list like: `password` and `12345678`

## Related Issue

- https://github.com/owncloud/web/issues/9795

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] e2e tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -1256,6 +1256,7 @@ def ocisService(type, tika_enabled = False):
         "WEB_ASSET_PATH": "%s/dist" % dir["web"],
         "FRONTEND_SEARCH_MIN_LENGTH": "2",
         "FRONTEND_OCS_ENABLE_DENIALS": True,
+        "FRONTEND_PASSWORD_POLICY_BANNED_PASSWORDS_LIST": "/srv/config/drone/password-policy-banned-passwords.txt",
```
Comment (saw-jan) [inline .drone.star:1259]: or simply
```suggestion
        "FRONTEND_PASSWORD_POLICY_BANNED_PASSWORDS_LIST": "/srv/config/drone/banned-passwords.txt",
```

File: tests/e2e/cucumber/features/smoke/spaces/publicLink.feature [hunk 2]
```diff
@@ -70,3 +70,25 @@ Feature: spaces public link
...
+    And "Alice" uploads the following resources
+      | resource  |
+      | lorem.txt |
+    And "Alice" creates a public link for the resource "lorem.txt" using the sidebar panel
+    And "Alice" renames the most recently created public link of resource "lorem.txt" to "myPublicLink"
+    When "Alice" tries to sets the password of the public link named "myPublicLink" of resource "lorem.txt" to "<banned-password>"
+    Then "Alice" should see an error message
+    """
+    Unfortunately, your password is commonly used. please pick a harder-to-guess password for your safety
+    """
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/spaces/publicLink.feature:89]:
```suggestion
      """
      Unfortunately, your password is commonly used. please pick a harder-to-guess password for your safety
      """
```

File: tests/e2e/cucumber/features/smoke/spaces/publicLink.feature [hunk 3]
```diff
@@ -70,3 +70,25 @@ Feature: spaces public link
     When "David" opens the public link "imageLink"
     Then for "David" file "testavatar.jpg" should be selected
     And "David" logs out
+
+
+  Scenario Outline: add banned password for public link
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/spaces/publicLink.feature:75]: better to add steps as journey
```suggestion
  Scenario: add banned password for public link
```

File: tests/e2e/cucumber/steps/ui/links.ts [hunk 4]
```diff
@@ -74,6 +74,21 @@ When(
...
+  async function (
+    this: World,
+    stepUser: string,
+    linkName: string,
+    resource: string,
+    newPassword: string
+  ): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const linkObject = new objects.applicationFiles.Link({ page })
+    await linkObject.addBannedPassword({ resource, linkName, newPassword })
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/links.ts:88]:
```suggestion
    await linkObject.setPassword({ resource, linkName, newPassword })
```

File: tests/e2e/support/objects/app-files/link/actions.ts [hunk 5]
```diff
@@ -192,10 +194,20 @@ export const addPassword = async (args: addPasswordArgs): Promise<void> => {
...
+
+  await setPassword(args)
   const message = await page.locator(linkUpdateDialog).textContent()
   expect(message.trim()).toBe('Link was updated successfully')
 }
 
+export const addBannedPassword = async (args: addPasswordArgs): Promise<void> => {
+  await setPassword(args)
+}
+
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/link/actions.ts:210]:
```suggestion
```

File: tests/e2e/support/objects/app-files/link/actions.ts [hunk 6]
```diff
@@ -175,7 +177,7 @@ export const changeName = async (args: changeNameArgs): Promise<string> => {
   return await getRecentLinkName(page)
 }
 
-export const addPassword = async (args: addPasswordArgs): Promise<void> => {
+const setPassword = async (args: addPasswordArgs): Promise<void> => {
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/link/actions.ts:180]:
```suggestion
const fillPassword = async (args: addPasswordArgs): Promise<void> => {
```

File: .drone.star [hunk 7]
```diff
@@ -1257,6 +1257,7 @@ def ocisService(type, tika_enabled = False):
         "FRONTEND_SEARCH_MIN_LENGTH": "2",
         "FRONTEND_OCS_ENABLE_DENIALS": True,
         "FRONTEND_AUTO_ACCEPT_SHARES": True,
+        "FRONTEND_PASSWORD_POLICY_BANNED_PASSWORDS_LIST": "/srv/config/drone/banned-passwords.txt",
```
Comment (saw-jan) [inline .drone.star:1260]:
```suggestion
        "FRONTEND_PASSWORD_POLICY_BANNED_PASSWORDS_LIST": "%s/tests/drone/password-policy-banned-passwords.txt" % dir["web"],
```

File: .drone.star [hunk 8]
```diff
@@ -1285,6 +1286,7 @@ def ocisService(type, tika_enabled = False):
                 "./ocis init",
                 "cp %s/tests/drone/app-registry.yaml /root/.ocis/config/app-registry.yaml" % dir["web"],
                 "./ocis server",
+                "cp %s/tests/drone/password-policy-banned-passwords.txt /srv/config/drone/banned-passwords.txt" % dir["web"],
```
Comment (saw-jan) [inline .drone.star:1289]:
```suggestion
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
this should work without having to copy file

File: tests/e2e/support/objects/app-files/link/index.ts [hunk 9]
```diff
@@ -55,6 +55,10 @@ export class Link {
     await this.#page.goto(startUrl)
   }
 
+  async setPassword(args: Omit<po.addPasswordArgs, 'page'>): Promise<void> {
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/link/index.ts:58]:
```suggestion
  async fillPassword(args: Omit<po.addPasswordArgs, 'page'>): Promise<void> {
```

File: .drone.star [hunk 10]
```diff
@@ -1257,6 +1257,7 @@ def ocisService(type, tika_enabled = False):
         "FRONTEND_SEARCH_MIN_LENGTH": "2",
         "FRONTEND_OCS_ENABLE_DENIALS": True,
         "FRONTEND_AUTO_ACCEPT_SHARES": True,
+        "FRONTEND_PASSWORD_POLICY_BANNED_PASSWORDS_LIST": "%s/tests/drone/password-policy-banned-passwords.txt" % dir["web"],
```
Comment (saw-jan) [inline .drone.star:1260]:
```suggestion
        "FRONTEND_PASSWORD_POLICY_BANNED_PASSWORDS_LIST": "%s/tests/drone/banned-passwords.txt" % dir["web"],
```

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

File: tests/e2e/cucumber/steps/ui/links.ts [hunk 11]
```diff
@@ -136,6 +151,17 @@ Then(
   }
 )
 
+Then(
+  '{string} should see an error message',
+  async function (this: World, stepUser: any, errorMessage: string): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const linkObject = new objects.applicationFiles.Link({ page })
+    const actualErrorMessage = await linkObject.checkErrorMessage()
+    expect(errorMessage).toBe(actualErrorMessage)
```
Comment (SagarGi) [inline tests/e2e/cucumber/steps/ui/links.ts:160]:
```suggestion
    expect(actualErrorMessage).toBe(errorMessage)
```

**Review by SagarGi — APPROVED** [review]
Other LGTM :+1:

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**